go/tools/builders/compilepkg.go

```diff
@@ -343,6 +342,15 @@ func compileArchive(
 		if err != nil {
 			return err
 		}
+		if coverMode != "" {
+			// Compile original source files, not coverage instrumented, for nogo
```

Comment:
Do we need this `cgo2` call if nogo isn't used? 

It looks like we are now compiling cgo sources twice - is this true and if so, is it really necessary?

Reply:
@emmaxy Friendly ping, I would call this the last blocker to merging this.

Reply:
Good call, thanks! This additional cgo compile is not needed if nogo doesn't run, I updated the condition accordingly.